src/components/Table/TableRow.tsx

```diff
@@ -8,9 +9,27 @@ export default function TableRow(props: RowRendererProps<any>) {
     return (
       <Fragment key={props.row.id}>
         <OutOfOrderIndicator top={props.top} height={props.height} />
-        <Row {...props} />
+        <ContextMenu>
+          <Row {...props} />
+        </ContextMenu>
       </Fragment>
     );
 
-  return <Row {...props} />;
+  return (
+    <ContextMenu>
+      <Row {...props} />
+    </ContextMenu>
+  );
 }
+
+const ContextMenu = (props: any) => {
+  const { contextMenuRef }: any = useProjectContext();
+  function handleClick(e: any) {
+    e.preventDefault();
+    const input = e?.target as HTMLElement;
+    if (contextMenuRef?.current) {
+      contextMenuRef?.current?.setAnchorEl(input);
+    }
+  }
+  return <span onContextMenu={(e) => handleClick(e)}>{props.children}</span>;
+};
```

Comment:
I don’t think `ContextMenu` needs to be a separate component. I just set the `onContextMenu` prop on `Row` and the `e.target` is the cell `div`, which is the same in both methods.

Reply:
Good catch, my dude! The less code the better. I will update the changes